style-guide.md

```diff
@@ -395,6 +395,36 @@ do).
 * `compute_` **should** be used for cases where something is being computed.
 * `fetch_` **should** be used for retrievals that are external, such as via an HTTP request, or other IO.
 
+## General comments on naming
+
+Strive to choose names that clearly communicate intent; you want to essentially use the "smallest" concept you can while still transmitting the relevant information to future readers (including yourself).
+
+Of particular importance is naming things based on what they provide, not what they do or details of how they do it. First an example, then further rationale:
+
+```python
+# bad
+## implies a package name of `enum` that doesn't tell
+## the consumer of a codebase much about what is inside.
+from enum.signature_enum import SignatureEnum
+
+from datastructures.header_helpers import HeaderCounter
+
+s = SignatureEnum.some_application_domain()
+h = HeaderCounter()
+# use s an h...
+
+
+
+# better
+from signatures.domains import SignatureDomain
+from headers.counter import HeaderCounter
+
+s = SignatureDomain.for_some_application()
+h = HeaderCounter()
+# use s an h...
```

Comment:
Agree with everything except for implicitly endorsing single-letter variable names :)

Reply:
good call